nixos/modules/services/networking/wgautomesh.nix

```diff
@@ -0,0 +1,149 @@
+{ lib, config, pkgs, ... }:
+with lib;
+let
+  cfg = config.services.wgautomesh;
+  gossipSecretFile = "/var/lib/wgautomesh/gossip_secret";
```

Comment:
IIUC, this file is generated and placed to the right place **before** wgautomesh service starts.

If so, I suggest to use systemd's `LoadCredential` for it.  Here [is][exp1] [an][exp2] [example][exp3].

If `wgautomesh` cannot have environment variable in a file path in its config file, such as `gossip_secret_file = 
 $CREDENTIALS_DIRECTORY/gossip_secret`, then you also need to substitute the environment variable `CREDENTIALS_DIRECTORY` with its value in `ExecStartPre` using [`envsubst`][sub-example].

[exp1]: https://github.com/NixOS/nixpkgs/blob/f89354c12f9824c764036887d0fef905c26a81af/nixos/modules/services/matrix/dendrite.nix#L75-L86
[exp2]: https://github.com/NixOS/nixpkgs/blob/f89354c12f9824c764036887d0fef905c26a81af/nixos/modules/services/matrix/dendrite.nix#L290
[exp3]: https://github.com/NixOS/nixpkgs/blob/f89354c12f9824c764036887d0fef905c26a81af/nixos/modules/services/matrix/dendrite.nix#L100-L113
[sub-example]: https://github.com/NixOS/nixpkgs/blob/f89354c12f9824c764036887d0fef905c26a81af/nixos/modules/services/matrix/dendrite.nix#L291-L295